clientv3/client.go

```diff
@@ -157,16 +158,22 @@ func (c *Client) Dial(endpoint string) (*grpc.ClientConn, error) {
 		grpc.WithBlock(),
 		grpc.WithTimeout(c.cfg.DialTimeout),
 	}
-	if c.creds != nil {
-		opts = append(opts, grpc.WithTransportCredentials(*c.creds))
-	} else {
-		opts = append(opts, grpc.WithInsecure())
-	}
 
 	proto := "tcp"
-	if url, uerr := url.Parse(endpoint); uerr == nil && url.Scheme == "unix" {
-		proto = "unix"
-		// strip unix:// prefix so certs work
+	if url, uerr := url.Parse(endpoint); uerr == nil {
+		switch url.Scheme {
+		case "unix":
+			proto = "unix"
+		case "http":
+			c.creds = nil
+		case "https":
```

Comment:
i think if user provides https without creds, we should use the default empty tls config as https://github.com/coreos/etcd/blob/master/etcdctl/ctlv3/command/global.go#L142